### DISCUSSION TITLE
Override/extend properties locally

### DISCUSSION THREAD

**Discussion by twam** [devcontainers/community#28] [Q&A]
Hello,

Is there any way to override properties of a devcontainer defined in `devcontainer.json` locally for a user, e.g. by defining a file like `devcontainer-override.json` in the same directory or a global one in the home directory?

My use-case is that I'd like to extend the mounts in my containers to my local needs and include stuff like `.zshrc` without touching the `devcontainer.json` which is git-versioned.

I searched a lot around, but couldn't find any hints that something like this is possible or if so, how.

Best regards
  Tobias

**Comment by Chuxel** [comment]
There is not today. However, the combination of allowing devcontainer.json to exist in sub-folders (e.g. .devcontainer/user/devcontainer.json) plus an extends keyword could enable something like this.  https://github.com/devcontainers/spec/issues/22 tracks the extends part of this.
